app/validators/qc_receptions_factory_validator.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# Failed validations return unprocessable_entity
+# These are being validated before any entity is created
+class QcReceptionsFactoryValidator < ActiveModel::Validator
+  def validate(record)
+    validations = %i[validate_qc_results_list validate_assay_type]
+
+    validations.each do |validation|
+      next if record.errors.present?
+
+      send(validation, record)
+    end
+  end
+
+  def validate_qc_results_list(record)
+    if record.qc_results_list.empty?
+      record.errors.add :qc_results_list, 'Is empty'
+      return
+    end
+    validate_each(record)
+  end
+
+  def validate_each(record)
+    is_empty = true
+    record.qc_results_list.each do |qc_hash|
```

Comment:
There are methods to deal with this https://ruby-doc.org/core-3.0.2/Enumerable.html#method-i-all-3F

Reply:
Done